lib/sww.js

```diff
@@ -52,13 +53,28 @@ ServiceWorkerWare.prototype.handleEvent = function sww_handleEvent(evt) {
 
 ServiceWorkerWare.prototype.onFetch = function sww_onFetch(evt) {
   var steps = this.router.match(evt.request.method, evt.request.url);
+
+  // Push the fallback middleware at the end of the list if it's defined
+  // also being sure that wont be executed if there is a response already.
+  if (this.fallbackMw) {
+    var fallback = this.fallbackMw;
+    steps.push(function(req, res) {
+      if (res !== null) {
+        return Promise.resolve(res);
+      }
+
+      return fallback(req, res);
+    });
+  }
+
   if (steps.length === 0) {
-    // XXX: we should have at least 1 basic middle ware that we install
-    // and can be overwritten. So far ... go to the network :(
+    // If no fallback mw, we go to the network.
+    // XXX: We could setup a set of policies here too.
```

Comment:
Instead of checking for a default middleware, adding, then, if not available, adding your own default middleware. Convert `fetch(evt.request)` into the default middleware and set in during construction. Then you can simply push the already set fallback.